### PR TITLE
Refactor big and small datas handling

### PR DESCRIPTION
**Pull request by pbobyrev**

### CODE REVIEW

File: src/tests/unit/utils/test_utils.py [hunk 1]
```diff
@@ -48,13 +49,33 @@ def dummy_function(name):
...
-    (1e18, datetime(2001, 9, 9, 1, 46, 40)),
-    (0, datetime(1970, 1, 1, 0, 0)),
-    (-1e18, datetime(1938, 4, 24, 22, 13, 20))
+def test_datetime_to_timestamp():
+    test_cases = [
+        ("0001-01-01", -62135596800.0),
+        ("1970-01-01", 0),
+        ("2000-01-01", 946684800),
+        ("2023-01-01", 1672527600.0),
+        #("2023-01-01 00:00:00", 1672527600.0),
```
Comment (Anna050689) [inline src/tests/unit/utils/test_utils.py:58]: Please, uncomment these lines.

File: src/syngen/ml/utils/utils.py [hunk 2]
```diff
@@ -15,15 +16,42 @@
...
+def datetime_to_timestamp(dt):
+    try:
+        dt = parser.parse(dt).replace(tzinfo=None)
+        delta = dt - datetime(1970, 1, 1)
+        return delta.total_seconds()
+    except parser._parser.ParserError as e:
+        year = re.match("\d+", e.args[0][5:]).group(0)
+        if int(year) > 9999:
+            return 253402214400.0
+        elif int(year) < 1:
```
Comment (Anna050689) [inline src/syngen/ml/utils/utils.py:28]: Should we use the same value as in min_allowed_time_ms?

**Review by Anna050689 — CHANGES_REQUESTED** [review]

File: src/syngen/ml/utils/utils.py [hunk 3]
```diff
@@ -15,15 +16,44 @@
...
+
+def timestamp_to_datetime(timestamp):
+    # Calculate the number of seconds in the UNIX epoch and the number of seconds left
+    max_allowed_time_ms = 253402214400
+    min_allowed_time_ms = -62135596800
+
+    if timestamp >= max_allowed_time_ms:
+        return datetime(9999, 12, 31, 23, 59, 59, 999999)
+    elif timestamp <= min_allowed_time_ms:
+        return datetime(1, 1, 1, 0, 0, 0, 0)
```
Comment (tdspora) [inline src/syngen/ml/utils/utils.py:42]: Added the limits to https://kb.epam.com/display/EPMCTDM/Known+Issues

**Review by tdspora — APPROVED** [review]
LGTM

**Review by Anna050689 — APPROVED** [review]

**Pull request merged**